moja-kuhinjica-frontend/src/components/modal/changePassword/ChangePasswordModal.tsx

```diff
@@ -0,0 +1,113 @@
+import { FormInput } from '@/components/input/FormInput'
+import { ErrorLabel } from '@/components/label/ErrorLabel'
+import React, { useState } from 'react'
+import { FieldValues, useForm } from 'react-hook-form'
+import Modal from 'react-modal'
+import { bgModal } from '../../../constants/constants'
+import styles from './ChangePasswordModal.module.scss'
+import passwordIcon from 'public/static/assets/images/password.svg'
+import UserService from '@/service/User.service'
+
+interface IChangePasswordModalProps {
+    modalIsOpen: boolean
+    closeModal: () => void
+}
+export const ChangePasswordModal = ({
+    modalIsOpen,
+    closeModal,
+}: IChangePasswordModalProps): JSX.Element => {
+    const [showError, setShowError] = useState<boolean>(false)
+    const [errorMessage, setErrorMessage] = useState<string>('')
+
+    const {
+        register,
+        handleSubmit,
+        reset,
+        formState: { errors },
+    } = useForm()
+
+    const validate = (data: FieldValues): void => {
+        if (data.newPassword === data.confirmPassword) {
+            delete data.confirmPassword
+            setShowError(false)
+            changePassword(data)
+            closeModal()
+        } else {
+            setErrorMessage('Šifre se ne poklapaju. Pokušajte ponovo.')
+            setShowError(true)
+        }
+    }
+
+    const changePassword = (data: FieldValues): void => {
+        UserService.changePassword(data)
+            .then((res) => {
+                // alert('successfully edited')
+                // reset()
```

Comment:
When we have an error we don't want modal to be closed, fix this please :)

Reply:
Done.